src/mkdocs_gallery/gen_single.py

```diff
@@ -788,9 +848,7 @@ def execute_code_block(compiler, block, script: GalleryScript):
 
     try:
         ast_Module = _ast_module()
-        code_ast = ast_Module([bcontent])
```

Comment:
I was surprised to see this but it is indeed present in sphinx-gallery: https://github.com/sphinx-gallery/sphinx-gallery/blob/master/sphinx_gallery/gen_rst.py#L965

They probably use this as a pre-validator of good syntax. Should we keep it too ? In that case I suggest to keep these original lines and simply add your `if _needs_async_handling` extra two lines below (dropping `_parse_code`). That way the change will be easier to spot when comparing with sphinx-gallery code

Reply:
> They probably use this as a pre-validator of good syntax. Should we keep it too ?

That is what we basically have right now. If we find a `SyntaxError` there, we need to check whether this comes from using async stuff and can't just exit the `try` branch pre-maturely. In the light of https://github.com/smarie/mkdocs-gallery/pull/90#discussion_r1434048801, we could just bubble up a `SyntaxError` if our async handling doesn't resolve it.

Reply:
I've implemented my suggested in 5e0e63693b5c951687d7f8519ec253a84d200ca8. Please have a look.